extension/EdFi.Ods.Extensions.LMSX/LMS-Harmonizer/1090-lms-assignment-submissions.sql

```diff
@@ -41,9 +42,52 @@ BEGIN
 		ON EDFISTUDENT.Id = lmsUser.EdFiStudentId
     WHERE lmsSubmission.SourceSystem = @SourceSystem
 
+	-- Here we are building missing submissions when they are not present
+	-- We do this to standardize what is shown in the visualizations
+	ALTER TABLE
+		#ALL_SUBMISSIONS
+	ALTER COLUMN
+		AssignmentSubmissionIdentifier
+	int NULL;
+
+
+	INSERT INTO #ALL_SUBMISSIONS
+	SELECT
+		edfisectionassociation.SectionIdentifier
+			+ '#' + lmsxassignment.AssignmentIdentifier
+			+ '#' + lmsstudent.SourceSystemIdentifier
+			as SourceSystemIdentifier,
+		NULL as AssignmentSubmissionIdentifier,
+		edfisectionassociation.StudentUSI,
+		lmsxassignment.AssignmentIdentifier,
+		submsisionstatusdescriptor.DescriptorId,
+		NULL as SubmissionDateTime,
+		NULL as EarnedPoints,
+		NULL as Grade,
+		GETDATE() as CreateDate,
+		GETDATE() as LastModifiedDate,
+		NULL AS DeletedAt
+
+	FROM edfi.StudentSectionAssociation edfisectionassociation
+	INNER JOIN lmsx.Assignment lmsxassignment
+		ON edfisectionassociation.SectionIdentifier = lmsxassignment.SectionIdentifier
+	INNER JOIN edfi.Student edfistudent
+		ON edfistudent.StudentUSI = edfisectionassociation.StudentUSI
+	INNER JOIN lms.LMSUser lmsstudent
+		ON lmsstudent.EdFiStudentId = edfistudent.Id
+	LEFT JOIN lmsx.AssignmentSubmission lmsxsubmission
+		ON lmsxsubmission.AssignmentIdentifier = lmsxassignment.AssignmentIdentifier
+			AND lmsxsubmission.StudentUSI = edfisectionassociation.StudentUSI
+	INNER JOIN edfi.Descriptor sourcesystemdesc
+		ON sourcesystemdesc.DescriptorId = lmsxassignment.LMSSourceSystemDescriptorId
+	INNER JOIN edfi.Descriptor submsisionstatusdescriptor
+		ON submsisionstatusdescriptor.Namespace = 'uri://ed-fi.org/edfilms/SubmissionStatusDescriptor/' + sourcesystemdesc.CodeValue
+		AND submsisionstatusdescriptor.CodeValue in ('missing', 'MISSING')
```

Comment:
I think the two joins to the Descriptor table can be simplified with one statement like this: (I did not test it carefully, so this might not be 100% right).

```sql
	CROSS APPLY (
        SELECT
            Descriptor.DescriptorId
        FROM
            edfi.Descriptor submsisionstatusdescriptor
		WHERE
            submsisionstatusdescriptor.Namespace = 'uri://ed-fi.org/edfilms/SubmissionStatusDescriptor/' + @SourceSystem
		AND
            submsisionstatusdescriptor.CodeValue = 'missing'
    ) as submsisionstatusdescriptor
```